src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -1600,6 +1602,145 @@ void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *
+      hasError = true;
+    }
+    else
+    {
+      QgsVectorLayer *srcLayer = nullptr;
+      bool owner;
+      QString error;
+      if ( dropUri.layerType == QLatin1String( "vector" ) )
+      {
+        // open the source layer
+        srcLayer = dropUri.vectorLayer( owner, error );
+      }
+      else
+      {
+        // unsupported
+        hasError = true;
+        continue;
+      }
+
+      if ( !srcLayer )
+      {
+        importResults.append( tr( "%1: %2" ).arg( dropUri.name, error ) );
+        hasError = true;
+        continue;
+      }
+
+      if ( srcLayer->isValid() )
+      {
+        uri = item->path();
+
+        // check if the destination layer already exists
+        bool exists = false;
+        const QVector< QgsDataItem * > c( item->children() );
+        for ( const QgsDataItem *child : c )
+        {
+          if ( child->name() == dropUri.name )
+          {
+            exists = true;
+          }
+        }
+
+        if ( ! exists || QMessageBox::question( nullptr, tr( "Overwrite Layer" ),
+                                                tr( "Destination layer <b>%1</b> already exists. Do you want to overwrite it?" ).arg( dropUri.name ), QMessageBox::Yes |  QMessageBox::No ) == QMessageBox::Yes )
+        {
+          QgsVectorLayer *vectorSrcLayer = qobject_cast < QgsVectorLayer * >( srcLayer );
+          QVariantMap options;
+          //  options.insert( QStringLiteral( "driverName" ), QStringLiteral( "GPKG" ) );
+          options.insert( QStringLiteral( "update" ), true );
+          options.insert( QStringLiteral( "overwrite" ), true );
+          options.insert( QStringLiteral( "layerName" ), dropUri.name );
+          QgsVectorLayerExporterTask *exportTask = new QgsVectorLayerExporterTask( vectorSrcLayer, uri, QStringLiteral( "ogr" ), vectorSrcLayer->crs(), options, owner );
+          mainTask->addSubTask( exportTask, importTasks );
+          importTasks << exportTask;
+          // when export is successful:
+          connect( exportTask, &QgsVectorLayerExporterTask::exportComplete, item, [ = ]()
+          {
+            notify( tr( "Import to database" ), tr( "Import was successful." ), context, Qgis::MessageLevel::Success );
+            item->refresh();
+          } );
+
+          // when an error occurs:
+          connect( exportTask, &QgsVectorLayerExporterTask::errorOccurred, item, [ = ]( Qgis::VectorExportResult error, const QString & errorMessage )
+          {
+            if ( error != Qgis::VectorExportResult::UserCanceled )
+            {
+              QgsMessageOutput *output = QgsMessageOutput::createMessageOutput();
+              output->setTitle( tr( "Import to database" ) );
+              output->setMessage( tr( "Failed to import some vector layers!\n\n" ) + errorMessage, QgsMessageOutput::MessageText );
+              output->showMessage();
+            }
+          } );
+        } // do not overwrite
```

Comment:
Kill those end of conditional block comments? I prefer the ones (that exists) at the beginning of the conditions :)